tests/playwright-test/playwright.artifacts.spec.ts

```diff
@@ -189,6 +189,35 @@ test('should work with screenshot: only-on-failure', async ({ runInlineTest }, t
   ]);
 });
 
+test('should work with screenshot: only-on-failure & fullPage', async ({ runInlineTest, server }, testInfo) => {
+  const result = await runInlineTest({
+    'artifacts.spec.ts': `
+    const { test } = pwt;
+
+    test('should fail and take fullPage screenshots', async ({ page }) => {
+      await page.setViewportSize({ width: 500, height: 500 });
+      await page.goto('${server.PREFIX}/grid.html');
+      expect(1).toBe(2);
+    });
+    `,
+    'playwright.config.ts': `
+      module.exports = { use: { screenshot: { mode: 'only-on-failure', fullPage: true } } };
+    `,
+  }, { workers: 1 });
+  expect(result.exitCode).toBe(1);
+  expect(result.passed).toBe(0);
+  expect(result.failed).toBe(1);
+  expect(listFiles(testInfo.outputPath('test-results'))).toEqual([
+    'artifacts-should-fail-and-take-fullPage-screenshots',
+    '  test-failed-1.png',
+    'report.json',
+  ]);
+  const screenshotFailure = fs.readFileSync(
+      testInfo.outputPath('test-results', 'artifacts-should-fail-and-take-fullPage-screenshots', 'test-failed-1.png')
+  );
+  expect.soft(screenshotFailure).toMatchSnapshot('screenshot-grid-fullpage.png');
```

Comment:
I think this snapshot file missing from the PR.

Reply:
True, sorry about that. Just added

Reply:
Good question, I forgot about that! Since our grid page is platform-independent, we can have just a single snapshot. However, to remove platform suffix from the name, you have to add `snapshotPathTemplate` to [this file](https://github.com/microsoft/playwright/blob/main/tests/playwright-test/playwright.config.ts):

```ts
const config: Config = {
  snapshotPathTemplate: '__screenshots__/{testFilePath}/{arg}{ext}',
};
```

This should produce a snapshot in a different location, without the platform suffix, so that we can run the new test on linux/win/mac CI.

Reply:
It seems that the `Config` type from `'./stable-test-runner'` (the one from [the file you mentioned](https://github.com/microsoft/playwright/blob/main/tests/playwright-test/playwright.config.ts)) does not have a `snapshotPathTemplate` parameter.

Reply:
That's right! I have [just rolled](https://github.com/microsoft/playwright/commit/d294c5da33dfe60bca2b76ca6893969002cbd62e) `stable-test-runner` to the new version and landed `snapshotPathTemplate`. You should be good after rebase.

Reply:
Thanks @dgozman ! I have just updated the screenshot